java-client/src/main/java/org/opensearch/client/opensearch/indices/CreateIndexRequest.java

```diff
@@ -304,6 +306,25 @@ public final Builder mappings(@Nullable TypeMapping value) {
             return this;
         }
 
+        /**
+         * Mapping for fields in the index. If specified, this mapping can include:
+         * <ul>
+         * <li>Field names</li>
+         * <li>Field data types</li>
+         * <li>Mapping parameters</li>
+         * </ul>
+         * <p>
+         * API name: {@code mappings}
+         */
+        public final Builder mappings(@Nullable String typeMappingJson) {
+            return this.mappings(deserialize(TypeMapping._DESERIALIZER, typeMappingJson));
+        }
+
+        private <T> T deserialize(JsonpDeserializer<T> deserializer, String json) {
+            JsonpMapper jsonpMapper = new JacksonJsonpMapper();
```

Comment:
This is not going to work, the `JsonpMapper` is provided by the user when `OpenSearchClient` instance is created, it has to be respected.